resources/sdk/purecloudjavascript/templates/configuration.mustache

```diff
@@ -0,0 +1,167 @@
+const ConfigParser = require('configparser');
+const os = require('os');
+const path = require('path');
+const fs = require('fs');
+
+import Logger from './logger.js';
+
+class Configuration {
+	/**
+	 * Singleton getter
+	 */
+	get instance() {
+		return Configuration.instance;
+	}
+
+	/**
+	 * Singleton setter
+	 */
+	set instance(value) {
+		Configuration.instance = value;
+	}
+
+	constructor() {
+		if (!Configuration.instance) {
+			Configuration.instance = this;
+		}
+
+		this.configPath = typeof window !== 'undefined' ? '' : path.join(os.homedir(), '.genesyscloudjavascript', 'config');
+		this.refresh_access_token = true;
+		this.refresh_token_wait_max = 10;
+		this.live_reload_config = true;
+		this.host;
+		this.environment;
+		this.basePath;
+		this.authUrl;
+		this.config;
+		this.logger = new Logger();
+		this.setEnvironment();
+		this.liveLoadConfig();
+	}
+
+	liveLoadConfig() {
+		// If in browser, don't read config file, use default values
+		if (typeof window !== 'undefined') {
+			this.configPath = '';
+			return;
+		}
+
+		this.updateConfigFromFile();
+
+		if (this.live_reload_config && this.live_reload_config === true) {
+			try {
+				const watcher = fs.watchFile(this.configPath, { persistent: false, interval: 30000 }, (eventType, filename) => {
+					this.updateConfigFromFile();
+					if (!this.live_reload_config) {
+						watcher.close();
+					}
+				});
+			} catch (err) {
+				// do nothing
+			}
+		}
+	}
+
+	setConfigPath(path) {
+		if (path) {
+			this.configPath = path;
+		}
+	}
+
+	updateConfigFromFile() {
+		var configparser = new ConfigParser();
+
+		try {
+			configparser.read(this.configPath); // If no error catched, indicates it's INI format
+			this.config = configparser;
+		} catch (error) {
+			if (error.name && error.name === 'MissingSectionHeaderError') {
+				// Not INI format, see if it's JSON format
+				var configData = fs.readFileSync(this.configPath, 'utf8');
+				this.config = {
+					_sections: JSON.parse(configData), // To match INI data format
+				};
+			}
+		}
+
+		if (this.config) this.updateConfigValues();
+	}
+
+	updateConfigValues() {
```

Comment:
I've got a weird issue where most of the time when the config is JSON booleans won't update

Reply:
Turns out it's a simple but nasty bug there. Fixed